clients/src/main/java/org/apache/kafka/common/record/EosLogEntry.java

```diff
@@ -197,26 +208,19 @@ protected LogRecord makeNext() {
     }
 
     public void setOffset(long offset) {
-        buffer.putLong(OFFSET_OFFSET, offset);
+        buffer.putLong(BASE_OFFSET_OFFSET, offset);
     }
 
-    public void setCreateTime(long timestamp) {
-        long currentTimestamp = maxTimestamp();
+    @Override
+    public void setTimestamp(TimestampType timestampType, long maxTimestamp) {
```

Comment:
I considered it. It definitely makes sense for `EosLogEntry`, but less so for `OldLogEntry`. If the old message is uncompressed, then we're really setting the record timestamp here (not the max timestamp of a record set). That said, I guess `setMaxTimestamp` is still technically correct even in that case, and maybe we should be emphasizing making the new API as natural as possible. Let's go ahead and change it.